src/docs/03-concepts/05-topology.md

```diff
@@ -17,7 +17,12 @@ Note that both types of :worker:workers: as well as external clients are roles a
 
 ![Cadence Service](/img/overview.png)
 
-At the core of Cadence is a highly scalable multitentant service. The service exposes all of its functionality through a strongly typed [gRPC API](https://github.com/uber/cadence-idl/tree/master/proto/uber/cadence/api/v1).
+At the core of Cadence is a highly scalable multitentant service. The service exposes all of its functionality through a strongly typed [gRPC API](https://github.com/uber/cadence-idl/tree/master/proto/uber/cadence/api/v1). A Cadence cluster include multiple services, each of which may run on multiple nodes for scalability and reliablity:
+- Front End (FE): which is a stateless service used to handle incoming requests from Workers. It is expected that an external load balancing mechanism is used to distribute load between Front End instances.
+- History Service (HS): where the core logic of orchestrating workflow steps and activities is implemented
+- Matching Service (MS): matches workflow/activity tasks that need to be executed to workflow/activity workers that are able to execute them. Matching is assigned task for execution by the history service
+- Worker Service (WS - not shown on diagram): implements Cadence workflows and activities for internal requirements such as archiving
+- Workers: are effectively the client apps for Cadence this is where user create workflow and activity logic is executed
```

Comment:
Nit: `user created workflow`? 
Also needs a `.` between `Cadence this`

Reply:
Thanks. Fixed now.